JSON.md

```diff
@@ -0,0 +1,200 @@
+---
+RFC: 2
+Status: Proposed
+---
+
+# ValKey JSON Module RFC
+
+## Abstract
+
+The proposed ValKey JSON module supports the native JavaScript Object Notation (JSON) format to encode complex datasets inside 
+OSS ValKey. It is compliant with [RFC7159](http://www.ietf.org/rfc/rfc7159.txt) and [ECMA-404](http://www.ecma-international.org/publications/standards/Ecma-404.htm) 
+JSON data interchange standard. With this feature, users can natively store, query, and modify JSON data structures in 
+ValKey using the popular [JSONPath query language](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html). 
+The module should be API-compatible and RDB-compatible with Redis Ltd.’s RedisJSON v2.
+
+## Motivation
+
+JSON format is a widely used data exchange format and simplifies the development of applications that store complex data 
+structures by providing powerful searching and filtering capabilities. However, [ValKey core](https://github.com/valkey-io/valkey) 
+does not have a native data type for JSON. Redis Ltd.‘s RedisJSON is a popular Redis modules, but not under a true 
+open source license and hence cannot be distributed freely with ValKey. There’s a demand in the ValKey 
+community to have a JSON module that matches most of the features of RedisJSON and is as API-compatible as possible. 
+See the community discussions [here](https://github.com/orgs/valkey-io/discussions?discussions_q=is%3Aopen+JSON).
+
+## Design Considerations
+
+ValKey JSON module introduces a new JSON data type for ValKey, and commands to insert, update, delete and query JSON data. 
+The module should be API-compatible and RDB-compatible with Redis Ltd.’s RedisJSON v2. Users can natively store, query, 
+and modify JSON data structures, either wholly or partially, using the popular [JSONPath query language](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html).
+
+![](./images/json_overview.png)
+
+### Module OnLoad
```

Comment:
This section and the following few sections (including Keyspace Event Notification) simply specify the behaviour without any discussion about why, etc. It's better to move it to Specification.

Reply:
Moved to Specification.